DRY up dataGenerator and rm cruft from dbLoader

elocowan opened the pull request.

elocowan committed:
DRY up dataGenerator and rm cruft from dbLoader

AngieD8615 commented on dataGenerator.js:
```diff
@@ -88,182 +52,36 @@ while (numberOfReviews < numberToWrite) {
   for (var i = 0; i <= randomReviewAmount; i++) {
```
for line 47 ( for whatever reason I can not comment on it): why do you need product_id with the starting index of 1? If it can be indexed at 0, that may be cleaner code.

AngieD8615 commented on dataGenerator.js:
```diff
@@ -88,182 +52,36 @@ while (numberOfReviews < numberToWrite) {
+    reviews += `${numberOfReviews + 1},${product_id + 1},${random(1, 6)},${date.toString().slice(0, 24)},${faker.random.words()},${faker.lorem.paragraph()},${(Math.random() >= .3)},${false},${faker.name.findName()},${faker.internet.email()},${"none"},${random(0, 30)}\n`
 
     // generate values for this review's characteristics + add to csv
     for (var j = 0; j < products[product_id].characteristics.length; j++) {
```
isn't products[product_id].characteristics.length = 4? can you either write 4  (with a short comment about about why 4) or set up a variable with a simpler name (charLength) so that this is more readable. Your future self will thank you.

AngieD8615 commented on dataGenerator.js:
```diff
@@ -88,182 +52,36 @@ while (numberOfReviews < numberToWrite) {
-    } else if (numberOfReviews / numberToWrite <= 1) {
-      reviews10 += `${numberOfReviews + 1},${product_id + 1},${random(1, 6)},${date.toString().slice(0, 24)},${faker.random.words()},${faker.lorem.paragraph()},${(Math.random() >= .3)},${false},${faker.name.findName()},${faker.internet.email()},${"none"},${random(0, 30)}\n`
-    }
+    reviews += `${numberOfReviews + 1},${product_id + 1},${random(1, 6)},${date.toString().slice(0, 24)},${faker.random.words()},${faker.lorem.paragraph()},${(Math.random() >= .3)},${false},${faker.name.findName()},${faker.internet.email()},${"none"},${random(0, 30)}\n`
```
I feel like I am missing why you need the "+1"s. What is your thinking here?

AngieD8615 commented on dataGenerator.js:
```diff
@@ -88,182 +52,36 @@ while (numberOfReviews < numberToWrite) {
     // generate photos for this review + add to csv
     let photoNumber = random(0, 8);
     for (var a = 0; a < photoNumber; a++) {
       let reviewPhoto = `${photoCounter},${numberOfReviews + 1},${faker.image.imageUrl()}\n`;
```
why do you need photoCounter? is it to provide a unique ID? If so, did Josh say that it would be faster to insert a unique ID on the DB side? I am not sure on this but I wonder if thing would be more efficient if you omit the counters here and put them in on the database side...

AngieD8615 commented on dbLoader.js:
```diff
@@ -11,32 +11,25 @@ const pool = new Pool({
-
+let startReviewsTruncateTime;
+let endPhotosLoad;
 const loadReviews = () => {
```
Love the ES6, so clean!!

AngieD8615 commented on dbLoader.js:
```diff
@@ -11,32 +11,25 @@ const pool = new Pool({
-    .then(() => copyReviews())
+    .then(() => {
+      endReviewsTruncateTime = new Date;
+      console.log("\nreviews truncation took", (endReviewsTruncateTime - startReviewsTruncateTime) + 'ms');
```
Great use of logging time!

AngieD8615 commented on dbLoader.js:
```diff
@@ -48,72 +41,87 @@ const loadReviews = () => {
 loadCharacteristics();
-loadPhotos();
\ No newline at end of file
+loadPhotos();
```
Leave a single blank line at the end

AngieD8615 reviewed (approved):
Looks so much better!!!! Now I would feel good bragging about this awesome piece of work!

elocowan committed:
Connect client to 10M database

Legacy client now displays reviews from database.
Parallel to this commit, I also indexed two columns in the database to
speed up query times. product_id in the reviews table and review_id in
the reviews_photo table were indexed using hash, and now the queries are
blazing fast.

elocowan committed:
Connected characteristics route

elocowan committed:
Set up put routes for helpful and report

elocowan committed:
Fix issue with loading characteristics

MVP wet solution for adding characteristics, now that the number of
characteristics per product is no longer fixed.
